### PR TITLE
fix: fix schema instances in multiple kcl mods.

### PR DESCRIPTION
**Pull request by zong-zhe**
#### 1. Does this PR affect any open issues?(Y/N) and add issue references (e.g. "fix #123", "re #123".):

- [ ] N
- [x] Y 

fix issue https://github.com/kcl-lang/kcl/issues/636

#### 2. What is the scope of this PR (e.g. component or file name):

kclvm/compiler/src/codegen/llvm/context.rs
kclvm/compiler/src/codegen/llvm/node.rs

#### 3. Provide a description of the PR(e.g. more details, effects, motivations or doc link):

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Other

#### 4. Are there any breaking changes?(Y/N) and describe the breaking changes(e.g. more details, motivations or doc link):

- [x] N
- [ ] Y 

#### 5. Are there test cases for these changes?(Y/N) select and add more details, references or doc links:

- [x] Unit test
- [ ] Integration test
- [ ] Benchmark (add benchmark stats below)
- [ ] Manual test (add detailed scripts or steps below)
- [ ] Other

### CODE REVIEW

File: kclvm/config/src/modfile.rs [hunk 1]
```diff
@@ -84,10 +89,110 @@ pub struct KCLModFileExpectedSection {
...
+        self.entries.push_back(Entry::new(name, path));
+    }
+
+    /// [`contains_pkg_name`] will return [`Option::Some`] if there is an [`Entry`] with the given name in [`Entries`].
+    pub fn contains_pkg_name(&self, name: &String) -> Option<&Entry> {
+        self.entries.iter().find(|entry| entry.name() == name)
+    }
+
+    /// [`iter`] will return an iterator of [`Entry`] in [`Entries`].
+    pub fn iter(&self) -> std::collections::vec_deque::Iter<Entry> {
```
Comment (Peefy) [inline kclvm/config/src/modfile.rs:114]: Considering use `impl Deref` and `[#inline]`

File: kclvm/cmd/src/tests.rs [hunk 2]
```diff
@@ -458,6 +459,46 @@ fn test_compile_two_kcl_mod() {
...
+    let test_case_path = PathBuf::from("./src/test_data/instances");
+    let matches = app().arg_required_else_help(true).get_matches_from(&[
+        ROOT_CMD,
+        "run",
+        "-Y",
+        &test_case_path.join(kcl_yaml_path).display().to_string(),
+    ]);
+
+    let mut buf = Vec::new();
+    run_command(matches.subcommand_matches("run").unwrap(), &mut buf).unwrap();
```
Comment (Peefy) [inline kclvm/cmd/src/tests.rs:491]: Windows CI failed

File: kclvm/driver/src/test_data/test_vendor/.kpm/config/kpm.json [hunk 3]
```diff
@@ -0,0 +1 @@
+{"DefaultOciRegistry":"ghcr.io","DefaultOciRepo":"kcl-lang","DefaultOciPlainHttp":false}
```
Comment (Peefy) [inline kclvm/driver/src/test_data/test_vendor/.kpm/config/kpm.json:1]: Should it be removed after testing is completed?

File: kclvm/parser/src/entry.rs [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+                    &kclvm_ast::MAIN_PKG.to_string(),
+                    opts,
+                )?);
+            }
+            return Ok(());
+        })?;
+        return Ok(result);
+    }
+
+    // Filling the kcl program files (*.k) in each entry.
```
Comment (Peefy) [inline kclvm/parser/src/entry.rs:306]: Unuseful comments.

File: kclvm/parser/src/entry.rs [hunk 5]
```diff
@@ -0,0 +1,413 @@
...
+/// ```
+pub fn get_compile_entries_from_paths(
+    file_paths: &[String],
+    opts: &LoadProgramOptions,
+) -> Result<Entries, String> {
+    if file_paths.is_empty() {
+        return Err("No input KCL files or paths".to_string());
+    }
+    let mut result = Entries::default();
+    let mut m = std::collections::HashMap::<String, String>::new();
```
Comment (Peefy) [inline kclvm/parser/src/entry.rs:212]: Is there a problem not using `IndexMap` here?  What does `m` mean?

File: kclvm/parser/src/entry.rs [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+                &root,
+                &kclvm_ast::MAIN_PKG.to_string(),
+                opts,
+            )?);
+            result.push_entry(entry);
+        }
+    }
+
+    if m.is_empty() {
+        m.insert("".to_string(), "".to_string());
```
Comment (Peefy) [inline kclvm/parser/src/entry.rs:261]: What does the statement mean?

File: kclvm/parser/src/entry.rs (hunk 5 above)
Comment (zong-zhe) [reply kclvm/parser/src/entry.rs]: removed it.

File: kclvm/parser/src/entry.rs (hunk 6 above)
Comment (zong-zhe) [reply kclvm/parser/src/entry.rs]: This is the code debt before, I copied. I removed it.

File: kclvm/parser/src/entry.rs (hunk 4 above)
Comment (zong-zhe) [reply kclvm/parser/src/entry.rs]: fixed.

File: kclvm/driver/src/test_data/test_vendor/.kpm/config/kpm.json (hunk 3 above)
Comment (zong-zhe) [reply kclvm/driver/src/test_data/test_vendor/.kpm/config/kpm.json]: fixed.

File: kclvm/cmd/src/tests.rs (hunk 2 above)
Comment (zong-zhe) [reply kclvm/cmd/src/tests.rs:491]: fixed.

File: kclvm/utils/src/path.rs [hunk 7]
```diff
@@ -87,3 +87,15 @@ fn test_adjust_canonicalization1() {
         path.display().to_string()
     );
 }
+
+pub fn is_dir(path: &str) -> bool {
```
[the remark this answers is not in the document]
Comment (zong-zhe) [reply kclvm/utils/src/path.rs:92]: fixed.

**Review by Peefy — APPROVED** [review]
LGTM

**Pull request merged**